types/validator_set.go

```diff
@@ -79,6 +79,24 @@ func NewValidatorSet(valz []*Validator) *ValidatorSet {
 	return vals
 }
 
+func (vals *ValidatorSet) ValidateBasic() error {
+	if vals.IsNilOrEmpty() {
+		return errors.New("validator set is nil or empty")
+	}
+
+	for _, val := range vals.Validators {
+		if err := val.ValidateBasic(); err != nil {
+			return fmt.Errorf("validator failed validate basic in set, error: %w", err)
```

Comment:
would be good to output which one failed

```suggestion
			return fmt.Errorf("invalid validator #%d: %w", i, err)
```